cocos/core/director.ts

```diff
@@ -731,6 +741,27 @@ export class Director extends EventTarget {
         }
     }
 
+    private buildRenderPipeline () {
+        if (this._root) {
+            this._root.customPipeline.beginSetup();
+            this._pipelineBuilder!.setup(this._root.cameraList, this._root.customPipeline);
+            this._root.customPipeline.endSetup();
+        }
+    }
+
+    private setupRenderPipelineBuilder () {
+        if (this._root && this._root.usesCustomPipeline
+            && legacyCC.internal.createCustomPipeline
+            && legacyCC.internal.customPipelineBuilderMap) {
+            const map: Map<string, PipelineBuilder> = legacyCC.internal.customPipelineBuilderMap;
+            this._pipelineBuilder = map.get(macro.CUSTOM_PIPELINE_NAME) || null;
+            if (this._pipelineBuilder === null) {
+                this._pipelineBuilder = map.get('Forward')!;
```

Comment:
Find required custom pipeline builder when customization is enabled. If pipeline not found, use builtin 'Forward' pipeline.